queue_job_chunk/models/queue_job_chunk.py

```diff
@@ -0,0 +1,55 @@
+
+from odoo import models, fields
+from odoo.addons.queue_job.job import job
+
+
+class QueueJobChunk(models.Model):
+    _name = "queue.job.chunk"
+
+    def _compute_reference(self):
+        for rec in self:
+            model = rec.model_id
+            record = rec.record_id
+            if model and record:
+                rec.reference = self.env[model].browse(record)
+
+    # component fields
+    usage = fields.Char("Usage")
+    collection = fields.Char("Collection")
+    apply_on_model = fields.Char("Apply on model")
+
+    data_original = fields.Many2one("ir.attachment", string="Original data")
```

Comment:
I will not add this data_original (teh origin can be a it_attachment, data from an api call....)
In the case of processing an attachment I will just put the attachment record in the reference field